From 9bc2b2438cc1648e94de93379670f49cc5077505 Mon Sep 17 00:00:00 2001
From: Yehonal <yehonal.azeroth@gmail.com>
Date: Wed, 21 Dec 2016 14:48:45 +0100
Subject: [PATCH 07/26] Fixed characters selection

---
 src/components/characters/index.jsx | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/components/characters/index.jsx b/src/components/characters/index.jsx
index 23aa36a..cf4783e 100644
--- a/src/components/characters/index.jsx
+++ b/src/components/characters/index.jsx
@@ -40,7 +40,10 @@ class CharactersScreen extends React.Component {
   }
 
   _onCharacterSelect(event) {
-    this.setState({ character: event.target.value });
+    this.setState({ 
+      index : event.target.value,
+      character: this.state.characters[event.target.value] 
+    });
   }
 
   _onJoin() {
@@ -74,11 +77,11 @@ class CharactersScreen extends React.Component {
 
           <form onSubmit={ this._onSubmit }>
             <fieldset>
-              <select value={ this.state.character }
+              <select value={ this.state.index }
                       onChange={ this._onCharacterSelect }>
-                { this.state.characters.map((character) => {
+                { this.state.characters.map((character, index) => {
                   return (
-                    <option key={ character.guid } value={ character }>
+                    <option key={index} value={ index }>
                       { character.name }
                     </option>
                   );
-- 
2.43.0

