From efde224ba81852de622017766b70b3cd6157232b Mon Sep 17 00:00:00 2001
From: Yehonal <yehonal.azeroth@gmail.com>
Date: Thu, 22 Dec 2016 13:40:57 +0100
Subject: [PATCH 15/26] small margin

---
 src/components/game/chat/index.styl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/components/game/chat/index.styl b/src/components/game/chat/index.styl
index 4a3ac9a..129a9ea 100644
--- a/src/components/game/chat/index.styl
+++ b/src/components/game/chat/index.styl
@@ -20,6 +20,7 @@ wowser .chat
 
   .message
     font-size: 14px
+    margin-bottom: 5px;
 
     &.system
       color: #FFCC00
-- 
2.43.0

